subserver_mgr.go

```diff
@@ -0,0 +1,399 @@
+package terminal
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightningnetwork/lnd/lncfg"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+)
+
+// subServerCfg defines the config values that must be set in order to properly
+// define a subServer.
+type subServerCfg struct {
+	// name is the string identifier of the sub-server.
+	name string
+
+	// remote must be set if the sub-server is running remotely and so must
+	// not be started in integrated mode.
+	remote bool
+
+	// startIntegrated defines how the sub-server should be started if it
+	// is to be run in integrated mode.
+	startIntegrated func(lnrpc.LightningClient, *lndclient.GrpcLndServices,
```

Comment:
What about lifting most of these closures out into an interface? would make it easier to implement test, and also wrap/implement some of the existing sub-servers with. 

It would also make it clearer imo exactly what a sub-server _is_ from the PoV of the manager.